src/kernel/mbrowse.ml

```diff
@@ -221,6 +221,22 @@ let of_signature sg =
   in
   [env, Browse_raw.Signature sg]
 
+let rec of_structure_items lst =
+  match lst with
+  | [] -> []
+  | (env,head) :: tl -> 
+    Browse_raw.Structure_item (head,env) :: (of_structure_items tl)
+    
+let rec of_signature_items lst =
+  match lst with
+  | [] -> []
+  | (env,head) :: tl -> 
+    Browse_raw.Signature_item (head,env) :: (of_signature_items tl)
+
```

Comment:
Same here. 
But if you want to keep the list reversed. I suggest (and also for `of_structure_item`): 

```ocaml
let of_structure_item = 
  let rec aux acc = function 
  | [] -> acc
  | (env, head) :: tl -> 
     aux (Browse_raw.Signature_item (head, env) :: acc)
  in aux []
```